src/core/Akka.Tests/Actor/TimerSpec.cs

```diff
@@ -0,0 +1,563 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="FSMTimingSpec.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2018 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Dispatch;
+using Akka.Event;
+using Akka.TestKit;
+using Akka.Util.Internal;
+using FluentAssertions;
+using Xunit;
+using static Akka.Actor.FSMBase;
+
+namespace Akka.Tests.Actor
+{
+    public class TimerSpec : AbstractTimerSpec
+    {
+        protected override Props TargetProps(IActorRef monitor, TimeSpan interval, bool repeat, Func<int> initial) =>
+            Props.Create(() => new Target(monitor, interval, repeat, initial ?? (() => 1)));
+    }
+
+    public class FsmTimerSpec : AbstractTimerSpec
+    {
+        protected override Props TargetProps(IActorRef monitor, TimeSpan interval, bool repeat, Func<int> initial) =>
+            Props.Create(() => new FsmTarget(monitor, interval, repeat, initial ?? (() => 1)));
+    }
+
+    public abstract class AbstractTimerSpec : AkkaSpec
+    {
+        int interval = 1;
+        TimeSpan dilatedInterval;
+
+        protected abstract Props TargetProps(IActorRef monitor, TimeSpan interval, bool repeat, Func<int> initial = null);
+
+        public AbstractTimerSpec()
+        {
+            dilatedInterval = Dilated(TimeSpan.FromSeconds(interval));
+        }
+
+        [Fact]
+        public void Must_schedule_non_repeated_ticks()
+        {
+            var probe = CreateTestProbe();
+            var _ref = this.Sys.ActorOf(TargetProps(probe.Ref, TimeSpan.FromMilliseconds(10), false));
+
+            probe.ExpectMsg<Tock>(m => m.N == 1);
+            probe.ExpectNoMsg(100);
+
+            _ref.Tell(End.Instance);
+            probe.ExpectMsg<GotPostStop>(m => m.TimerActive == false);
+        }
+
+        [Fact]
+        public void Must_schedule_repeated_ticks()
+        {
+            var probe = CreateTestProbe();
+            var _ref = this.Sys.ActorOf(TargetProps(probe.Ref, dilatedInterval, true));
```

Comment:
In general, try to avoid underscoring variables. A lot of code uses R# naming convention, and _-prefixed names are usually present in context of fields.

Reply:
renamed to `actor`, wanted to keep the name similar to scala version